### ISSUE TITLE
Configuration for shard iterator- LATEST, TRIM_HORIZON, AT_TIMESTAMP 

### ISSUE THREAD

**Issue by santoshkulk** [logstash-plugins/logstash-input-kinesis#19]
KCL provides an option to specify the initial position in the stream from where the application starts reading from. It can be one of LATEST, TRIM_HORIZON or AT_TIMESTAMP. I do not see a configuration option currently available in the logstash configuration to specify the same. Is there anything I am missing here?

**Comment by codekitchen** [comment]
Nope, it's just never come up so far. Right now it's hard-coded to TRIM_HORIZON: https://github.com/logstash-plugins/logstash-input-kinesis/blob/master/lib/logstash/inputs/kinesis.rb#L73

Making this configurable is certainly reasonable though.

**Comment by tschran** [comment]
I second this request.  

We have a number of streams with 7 day retention, so as we're setting up new logstash stream processors, each one has to chew through a week worth of data that we've already processed with our existing system.  It would be great to be able to tell it to just start processing from now and ignore the existing data.

**Comment by jennysivapalan** [comment]
We have the same use case @tschran so would love to see this feature.

**Comment by aloisbarreras** [comment]
hey @codekitchen I took a stab at adding support for LATEST [here](https://github.com/logstash-plugins/logstash-input-kinesis/pull/33).

I have never written ruby code in my life, so I'm sure it will need a little revision.

**Comment by jlaneve** [comment]
+1

**Comment by codekitchen** [comment]
`LATEST` is now supported thanks to a PR from @aloisbarreras 

`AT_TIMESTAMP` support involves further work if anybody is interested in tackling it -- most likely another config parameter for the timestamp. I'm not positive if logstash natively supports timestamp config vars or if we'd need to parse it out from a string in this plugin.
